src/components/organize/TagsManager/TagSelect.tsx

```diff
@@ -1,51 +1,147 @@
-import { Autocomplete } from '@material-ui/lab';
-import { TextField } from '@material-ui/core';
-import { useIntl } from 'react-intl';
+/* eslint-disable jsx-a11y/no-autofocus */
+import { Add } from '@material-ui/icons';
+import { useAutocomplete } from '@material-ui/lab';
+import { useState } from 'react';
+import {
+  Box,
+  List,
+  ListItem,
+  ListSubheader,
+  TextField,
+} from '@material-ui/core';
+import { FormattedMessage, useIntl } from 'react-intl';
 
+import { groupTags } from './utils';
+import { OnCreateTagHandler } from './types';
 import TagChip from './TagChip';
-import { ZetkinTag } from 'types/zetkin';
+import TagDialog from './TagDialog';
+import { ZetkinTag, ZetkinTagGroup } from 'types/zetkin';
 
 const TagSelect: React.FunctionComponent<{
   disabledTags: ZetkinTag[];
+  groups: ZetkinTagGroup[];
+  onCreateTag: OnCreateTagHandler;
   onSelect: (tag: ZetkinTag) => void;
   tags: ZetkinTag[];
-}> = ({ disabledTags, tags, onSelect }) => {
+}> = ({ disabledTags, groups, onCreateTag, onSelect, tags }) => {
   const intl = useIntl();
+
+  const [dialogOpen, setDialogOpen] = useState<{
+    defaultTitle: string;
+    open: boolean;
+  }>({
+    defaultTitle: '',
+    open: false,
+  });
+
+  const {
+    inputValue,
+    getInputProps,
+    getListboxProps,
+    getRootProps,
+    groupedOptions,
+  } = useAutocomplete({
+    getOptionLabel: (option) => option.title,
+    open: true,
+    options: tags,
+  });
+
+  const groupedFilteredTags = groupTags(
+    groupedOptions,
+    intl.formatMessage({
+      id: 'misc.tags.tagsManager.ungroupedHeader',
+    })
+  );
+
   return (
-    <Autocomplete
-      clearOnEscape
-      getOptionDisabled={(option) =>
-        disabledTags.map((tag) => tag.id).includes(option.id)
-      }
-      getOptionLabel={(option) => option.title}
-      groupBy={(option) =>
-        option.group?.title ||
-        intl.formatMessage({
-          id: 'misc.tags.tagsManager.ungroupedHeader',
-        })
-      }
-      onChange={(e, value) => {
-        if (value) {
-          onSelect(value);
-        }
-      }}
-      openOnFocus
-      options={tags}
-      renderInput={(params) => (
-        <TextField
-          {...params}
-          data-testid="TagsManager-tagSelectTextField"
-          placeholder={intl.formatMessage({
-            id: 'misc.tags.tagsManager.addTag',
-          })}
-          variant="outlined"
-        />
-      )}
-      renderOption={(option) => {
-        return <TagChip tag={option} />;
-      }}
-      style={{ width: 300 }}
-    />
+    <Box {...getRootProps()}>
+      <TextField
+        {...getInputProps()}
+        autoFocus
+        fullWidth
+        inputProps={{
+          'data-testid': 'TagManager-TagSelect-searchField',
+        }}
+        placeholder={intl.formatMessage({
+          id: 'misc.tags.tagsManager.addTag',
+        })}
+        variant="outlined"
+      />
+      {/* Options */}
```

Comment:
You could filter out tags that don't match the inputValue here. Like this:

![create-tag-filter](https://user-images.githubusercontent.com/4164774/166964792-b65eb7a5-85a1-4258-b09e-3b80be717b6b.gif)

Reply:
This works now. Thanks for testing